pkg/clients/dclient/discovery.go

```diff
@@ -98,75 +98,85 @@ func (c serverPreferredResources) GetServerVersion() (*version.Info, error) {
 
 // FindResource finds an API resource that matches 'kind'. If the resource is not
 // found and the Cache is not fresh, the cache is invalidated and a retry is attempted
-func (c serverPreferredResources) FindResource(apiVersion string, kind string) (*metav1.APIResource, schema.GroupVersionResource, error) {
-	r, gvr, err := c.findResource(apiVersion, kind)
+func (c serverPreferredResources) FindResource(groupVersion string, kind string) (*metav1.APIResource, schema.GroupVersionResource, error) {
+	r, gvr, err := c.findResource(groupVersion, kind)
 	if err == nil {
 		return r, gvr, nil
 	}
 
 	if !c.cachedClient.Fresh() {
 		c.cachedClient.Invalidate()
-		if r, gvr, err = c.findResource(apiVersion, kind); err == nil {
+		if r, gvr, err = c.findResource(groupVersion, kind); err == nil {
 			return r, gvr, nil
 		}
 	}
 
 	return nil, schema.GroupVersionResource{}, err
 }
 
-func (c serverPreferredResources) findResource(apiVersion string, kind string) (*metav1.APIResource, schema.GroupVersionResource, error) {
-	var serverResources []*metav1.APIResourceList
-	var err error
-	if apiVersion == "" {
-		serverResources, err = c.cachedClient.ServerPreferredResources()
```

Comment:
We always get `c.cachedClient.ServerGroupsAndResources()` instead of `c.cachedClient.ServerPreferredResources()`. `ServerPreferredResources` do not contain subresources by default.